stl/inc/xatomic.h

```diff
@@ -56,6 +56,15 @@ _STL_DISABLE_CLANG_WARNINGS
 _STD_BEGIN
 
 // ENUM memory_order
+#if _HAS_CXX20
+enum class memory_order : unsigned int { relaxed, consume, acquire, release, acq_rel, seq_cst };

```

Comment:
Previously, I believe that `enum memory_order`'s underlying type was chosen to be `int` (I haven't confirmed this, though). While `unsigned int` is more convenient, I believe that this is a theoretical ABI break, contrary to the paper's recommendation of

> In order to maintain binary compatibility the underlying type of the scoped enumeration type should be left unspecified, so that implementations can set it to match the underlying type that was previously chosen (either implicitly or explicitly) for the unscoped enumeration type.

That is, unusual code mentioning `underlying_type_t<memory_order>` would get different answers for C++17 vs. C++20 which we typically seek to avoid, so Standard versions can be freely mixed.

Can you confirm that `static_assert(is_same_v<underlying_type_t<memory_order>, int>)` was previously true? If so, I believe the least risky thing to do would be to:

1. Choose `int` as the explicitly specified underlying type in C++20,
2. Change the other code appropriately (e.g. the validity check would need to unconditionally cast both sides to `unsigned int`),
3. Consider marking this with a comment `// TRANSITION, ABI: unsigned int` if it would be desirable to change this when we can break ABI in the "vNext" release.

Reply:
> It has more than 5 P0s on the LWG reflector

From a quick review of the reflector thread, there are also a couple of plain P4s amid the sea of "P0 or P4"s. So it's likely that the PR will be applied in 10 or 12 years when we get around to looking at the issue.